test/2fa.test.js

```diff
@@ -0,0 +1,263 @@
+const dotenv = require('dotenv');
+dotenv.config('../.env');
+const assert = require('assert');
+const supertest = require('supertest');
+const models = require('../models');
+const MASTER_KEY_INFO = {
+    account_id: 'test.near',
+    secret_key: 'ed25519:2wyRcSwSuHtRVmkMCGjPwnzZmQLeXLzLLyED1NDMt4BjnKgQL6tF85yBx6Jr26D2dUNeC716RBoTxntVHsegogYw'
+};
+process.env = {
+    ...process.env,
+    ACCOUNT_CREATOR_KEY: JSON.stringify(MASTER_KEY_INFO),
+    WALLET_URL: 'https://wallet.nearprotocol.com',
+    NEW_ACCOUNT_AMOUNT: '500000001000000000000000000',
+    NODE_URL: 'https://rpc.ci-testnet.near.org'
+};
+const app = require('../app');
+
+const nearAPI = require('near-api-js');
+
+const ctx = {};
+const request = supertest(app.callback());
+
+jest.setTimeout(15000);
+
+beforeAll(async () => {
+    await models.sequelize.sync({ force: true });
+    ctx.near = await nearAPI.connect({
+        deps: { keyStore },
+        nodeUrl: process.env.NODE_URL
+    });
+});
+
+afterAll(async () => {
+    await models.sequelize.close();
+});
+
+beforeEach(() => {
+    ctx.savedLog = console.log;
+    ctx.logs = [];
+    console.log = (...args) => ctx.logs.push(args);
+});
+
+afterEach(() => {
+    console.log = ctx.savedLog;
+});
+
+const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+
+const twoFactorMethods = [
+    { kind: '2fa-email', detail: 'hello@example.com', publicKey: 'pkemail' },
+    { kind: '2fa-phone', detail: '+1 717 555 0101', publicKey: 'pkphone' },
+];
+
+const inMemorySigner = new nearAPI.InMemorySigner(keyStore);
+async function signatureFor(accountId, valid = true) {
+    let blockNumber = (await ctx.near.connection.provider.status()).sync_info.latest_block_height;
+    if (!valid) blockNumber = blockNumber - 101;
+    blockNumber = String(blockNumber);
+    const message = Buffer.from(blockNumber);
+    const signedHash = await inMemorySigner.signMessage(message, accountId);
+    const blockNumberSignature = Buffer.from(signedHash.signature).toString('base64');
+    return { blockNumber, blockNumberSignature };
+}
+
+const { parseSeedPhrase } = require('near-seed-phrase');
+const SEED_PHRASE = 'shoot island position soft burden budget tooth cruel issue economy destroy above';
+const keyPair = nearAPI.KeyPair.fromString(parseSeedPhrase(SEED_PHRASE).secretKey);
```

Comment:
any reason to not use just random key pair?

Reply:
was using what was in other examples